### PR TITLE
General Cargo docs, wip

### PR DESCRIPTION
**Pull request by insipx**
Docs + Bringing archive up to date with the latest substrate changes

incoming writeups on Storage indexing, memory issues, and architecture

### CODE REVIEW

File: substrate-archive-backend/src/frontend.rs [hunk 1]
```diff
@@ -123,36 +124,39 @@ const fn default_storage_mode() -> TransactionStorageMode {
...
-			wasm_runtime_substitutes,
 			// we do not support 'no_genesis', so this value is inconsiquential
 			no_genesis: false,
+			wasm_runtime_substitutes,
 		})
 	}
 }
 
+/// Main entry to initialize the substrate-archive backend client, used to
+/// call into the runtime of the network being indexed (e.g to execute blocks).
```
Comment (dvdplm) [inline substrate-archive-backend/src/frontend.rs:156]:
```suggestion
/// call into the runtime of the chain being indexed (e.g to execute blocks).
```

File: substrate-archive-backend/src/frontend/client.rs [hunk 2]
```diff
@@ -14,12 +14,14 @@
 // You should have received a copy of the GNU General Public License
 // along with substrate-archive.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Custom implementation of a Client for Archival use
+//! Custom implementation of a Client for Archival use.
+//! This client resembles the client in `client/service/src/client` except that it uses a read-only rocksdb
+//! database implementation, and the backend state trie access attempt to avoid caching like the substrate client.
+//! (Substrate Archive generally does not expect to access values multiple times during a run).
```
Comment (dvdplm) [inline substrate-archive-backend/src/frontend/client.rs:20]:
```suggestion
//! This client resembles the client in `client/service/src/client` except that it uses a read-only rocksdb
//! database implementation, and that the backend state trie aims at avoid caching as much as possible, unlike the normal substrate client.
//! (Substrate Archive generally does not expect to access values multiple times during a run).
```

**Review by dvdplm — APPROVED** [review]

**Review by maciejhirsz — APPROVED** [review]
Looks pretty clean!

**Review by TarikGul — APPROVED** [review]
👍

**Comment by dvdplm** [comment]
@insipx Do you know what's up with the CI failures?

**Comment by insipx** [comment]
Ah I think it's some leftover from updating substrate, i'll fix that asap

**Pull request merged**